airbyte-webapp/src/components/connection/ConnectionOnboarding/ConnectionOnboarding.tsx

```diff
@@ -0,0 +1,256 @@
+
+  return { sourceDefinitions, destinationDefinitions };
+};
+
+export const ConnectionOnboarding: React.FC<ConnectionOnboardingProps> = ({ onCreate }) => {
+  const { formatMessage } = useIntl();
+  const { sourceDefinitions, destinationDefinitions } = useConnectorSpecificationMap();
+
+  const [highlightedSource, setHighlightedSource] = useState<0 | 1 | 2 | 3>(1);
+  const [highlightedDestination, setHighlightedDestination] = useState<0 | 1 | 2 | 3>(0);
+
+  const sourceIds = useExperiment("connection.onboarding.sources", "").split(",");
+  const destinationIds = useExperiment("connection.onboarding.destinations", "").split(",");
+
+  const sources = useMemo(
+    () =>
+      DEFAULT_SOURCES.map(
+        (defaultId, index) => sourceDefinitions[sourceIds[index] || defaultId] ?? sourceDefinitions[defaultId]
+      ),
+    [sourceDefinitions, sourceIds]
+  );
+
+  const destinations = useMemo(
+    () =>
+      DEFAULT_DESTINATIONS.map(
+        (defaultId, index) =>
+          destinationDefinitions[destinationIds[index] || defaultId] ?? destinationDefinitions[defaultId]
+      ),
+    [destinationDefinitions, destinationIds]
+  );
+
+  const moreSourcesTooltip = formatMessage(
+    { id: "connection.onboarding.moreSources" },
+    { count: Math.floor(Object.keys(sourceDefinitions).length / 10) * 10 }
+  );
+
+  const moreDestinationsTooltip = formatMessage(
+    { id: "connection.onboarding.moreDestinations" },
+    { count: Math.floor(Object.keys(destinationDefinitions).length / 10) * 10 }
+  );
+
+  return (
+    <div className={styles.container}>
+      <Heading as="h2" size="lg" centered className={styles.heading}>
+        <FormattedMessage id="connection.onboarding.title" />
+      </Heading>
+      <div className={styles.connectors}>
+        <div className={styles.sources}>
+          <Text bold as="div" className={styles.sourcesTitle}>
+            <Tooltip
+              control={
+                <span>
+                  <FormattedMessage id="connection.onboarding.sources" /> <FontAwesomeIcon icon={faCircleQuestion} />
+                </span>
+              }
+            >
+              <FormattedMessage id="connection.onboarding.sourcesDescription" />
+            </Tooltip>
+          </Text>
+          {sources.map((source, index) => {
+            const tooltipText = formatMessage({ id: "connection.onboarding.addSource" }, { source: source?.name });
+            return (
+              <Tooltip
+                key={source?.sourceDefinitionId}
+                placement="right"
+                control={
+                  <button
+                    data-testid={`onboardingSource-${index}`}
+                    data-source-definition-id={source?.sourceDefinitionId}
+                    aria-label={tooltipText}
+                    className={styles.connectorButton}
+                    onClick={() => onCreate(source?.sourceDefinitionId)}
+                    onMouseEnter={() => setHighlightedSource(index as 0 | 1 | 2)}
```

Comment:
Still curious if this could be handled VIA css.

Reply:
Unfortunately not. Because of one stupid little thing, lack of z-index. We could actually solve in CSS the changing the highlighting of a line (leaving aside the gradient problematic I described above), but we need to make sure the highlighted line renders above the dashed ones. Since SVG doesn't support `z-index` and only render by order, we can't solve that via CSS unfortunately. Would be happen to be tought wrong, if you have specific code suggestions how to solve it via CSS.

Reply:
You've clearly done a deep dive here! I wonder if a `:hover` and a `:not:hover` could handle some of it. But this works and I don't think it's worth more dev time.

Reply:
Oh actually now I think again about it, I think we could handle it, if we don't switch the `fill` of the highlighted line, but have of each line the highlighted in a duplicate behind all the dotted ones, and just toggle `display: none` on those depending on the hover state, we can actually solve the z-index issue and the gradient can't be defined in CSS issue.

Only disadvantages would be that we'll need to have the illustration in DOM behind source AND destination buttons for it to work (but we can just use `order` in the flexbox to visually move them into the different positions again), as well as that we need to share CSS between ConnectionOnboarding and AirbyteIllustrations, since AirbyteIllustrations elements need to react need to react based on CSS classes in ConnectionOnboarding. But since I'd always prefer solving styling issues in CSS and not TS whenever possible I think the CSS solution would still be preferred and I'll go to switch to this. (Also helps us getting rid of the weird `button:disabled` is not triggering `onMouseEnter` in react.)

Reply:
Well glad I could help haha

Reply:
... But we'd still need this property, because we can't change the `mpath` `xlink:href` property via CSS, to adjust which path the dot is actually taking (or toggle `animateMotion`), so I'd need to fully copy the dots also 4 times, that feels like it's getting a bit much duplication just to use CSS over this TS code :-/

Reply:
![](https://media0.giphy.com/media/l44QxU2ntAiFKDnvq/giphy.gif)